django/contrib/auth/migrations/0010_update_proxy_permissions.py

```diff
@@ -0,0 +1,57 @@
+from django.contrib.auth import get_permission_codename
+from django.db import migrations
+
+
+def update_proxy_model_permissions(apps, schema_editor):
+    Permission = apps.get_model('auth', 'Permission')
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+
+    for Model in apps.get_models():
+        opts = Model._meta
+
+        if not opts.proxy:
+            continue
+
+        concrete_content_type = ContentType.objects.get_for_model(Model,
+                                                                  for_concrete_model=True)
+
+        default_proxy_codenames = [
+            get_permission_codename(action, opts)
+            for action in opts.default_permissions
+        ]
+        proxy_custom_codenames = {codename for codename, __ in opts.permissions}
+
+        concrete_custom_codenames = {
+            codename for codename, __ in opts.concrete_model._meta.permissions
+        }
+
+        custom_codenames = list(concrete_custom_codenames - proxy_custom_codenames)
+
+        codenames = default_proxy_codenames + custom_codenames
+
+        if opts.app_label == opts.concrete_model._meta.app_label:
+            proxy_content_type = ContentType.objects.get_for_model(
+                Model, for_concrete_model=False)
+
+            Permission.objects.filter(
+                codename__in=codenames,
+                content_type=concrete_content_type,
+            ).update(content_type=proxy_content_type)
+
+        else:
```

Comment:
This is the part where we cleanup the permissions where the `app_label` of the concrete model and the proxy model are different. Users with the old proxy permissions would now have access to admin pages they couldn't see before (but should have...). So, my idea is to just delete the old permissions and add a message to the release notes to let people know that they need to re-issue the permissions for any proxy model having a different `app_label` than their concrete model.
Other approaches:
- Do nothing: The old permissions will be listed along with the new ones and it might be confusing.
- Update the content type the same way we do it for concrete/proxy models with similar `app_label`: It might be surprising for users with permissions to all the sudden have access to pages they couldn't see before. We could also communicate that fact in the release notes...